apps/sync/test/unit/adapters/history_test.js

```diff
@@ -0,0 +1,174 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global
+  assert,
+  asyncStorage,
+  requireApp,
+  setup,
+  sinon,
+  suite,
+  teardown,
+  test,
+  SyncEngine
+*/
+
+var MockasyncStorage = {
```

Comment:
There are already a few asyncStorage mocks that you can reuse. There's actually one in shared that may be extended for your needs.

https://mxr.mozilla.org/gaia/find?text=&string=mock_async_storage